react-backend/client/src/components/visitors/qrcode.js

```diff
@@ -86,8 +88,13 @@ export class QRCode extends Component {
       activity: newActivity,
     };
 
+    const headers = new Headers({
```

Comment:
rather than rewrite this bit of code twice (here and line 126), you could put the headers in the class (rather than each method in the class), and then use them as `this.headers` in the fetch request. 

Would just mean a little less repeated code.